frontends/open-discussions/src/lib/api/api.js

```diff
@@ -23,9 +23,29 @@ import type {
 } from "../../flow/discussionTypes"
 import type { NotificationSetting } from "../../flow/settingsTypes"
 import type { SearchParams } from "../../flow/searchTypes"
+import _ from "lodash"
 
 export function search(params: SearchParams): Promise<*> {
-  const body = buildSearchQuery(params)
+  const searchParams = _.pick(params, [
+    "text",
+    "from",
+    "size",
+    "sort",
+    "channelName"
+  ])
+
+  if (params["facets"]) {
+    searchParams["activeFacets"] = Array.from(params["facets"]).reduce(
+      (obj, [facet, values]) => {
+        obj[facet] = values
+        return obj
+      },
+      {}
+    )
+    searchParams["activeFacets"].type = params["type"]
+  }
```

Comment:
Instead of the reduce, you can use [`Object.fromEntries`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries) for this:
```suggestion
    searchParams["activeFacets"] = {
       ...Object.fromEntries(params["facets"]),
       type: params["type"]
    }     
``` 
(It expects an iterable of key-value pairs, which maps satisfy.)

Reply:
Done!